frx_challenges/web/models.py

```diff
@@ -144,6 +144,24 @@ def ordered_results(self) -> list:
                 results_list.append(None)
         return results_list
 
+    @property
+    def percent_complete(self) -> str:
+        """
+        Return the percentage of completion of this evaluation
+        """
+        if self.status == Evaluation.Status.EVALUATED:
+            return "100%"
+        elif self.status == Evaluation.Status.EVALUATING:
+            if self.result:
+                percent_complete = self.result.get(
+                    "num_evals_done", 0
+                ) / self.result.get("total_evals", 1)
+                return f"{int(percent_complete * 100)}%"
+            else:
+                return "0%"
+        else:
+            return "0%"
```

Comment:
This won't work unfortunately. If you look at https://github.com/2i2c-org/frx-challenges/blob/main/frx_challenges/web/management/commands/evaluator.py#L199, you see that the `evaluation.result` is not saved until the evaluation is 100% complete.

```python
evaluation.status = Evaluation.Status.EVALUATED
evaluation.result = result
```

This will take a lot more logic in the evaluator to implement. I would suggest removing this component from this PR for now.

Reply:
Moved this to https://github.com/2i2c-org/frx-challenges/pull/245 with fixes to the evaluation object updates to gather results even if evaluation is still ongoing.